sky/resources.py

```diff
@@ -1621,9 +1655,18 @@ def _override_resources(
     @classmethod
     def _from_yaml_config_single(cls, config: Dict[str, str]) -> 'Resources':
 
-        resources_fields = {}
+        resources_fields: Dict[str, Any] = {}
+
+        # Extract infra field if present
+        infra = config.pop('infra', None)
+        resources_fields['infra'] = infra
+
+        # Keep backward compatibility with cloud, region, zone
         resources_fields['cloud'] = registry.CLOUD_REGISTRY.from_str(
             config.pop('cloud', None))
+        resources_fields['region'] = config.pop('region', None)
+        resources_fields['zone'] = config.pop('zone', None)
```

Comment:
I think `infra` should take priority if both are set.

Reply:
Only one of `infra` and `cloud/region/zone` can be specified. Otherwise, it will raise with the `__init__` function. Added a comment here.